transport-native-epoll/src/main/c/netty_epoll_native.c

```diff
@@ -346,6 +346,61 @@ static jint netty_epoll_native_sendmmsg0(JNIEnv* env, jclass clazz, jint fd, jbo
     return (jint) res;
 }
 
+static jint netty_epoll_native_recvmmsg0(JNIEnv* env, jclass clazz, jint fd, jboolean ipv6, jobjectArray packets, jint offset, jint len) {
+    struct mmsghdr msg[len];
+    memset(msg, 0, sizeof(msg));
+    struct sockaddr_storage addr[len];
+    int addrSize = sizeof(addr);
+    memset(addr, 0, addrSize);
+
+    int i;
+
+    for (i = 0; i < len; i++) {
+        jobject packet = (*env)->GetObjectArrayElement(env, packets, i + offset);
+        msg[i].msg_hdr.msg_iov = (struct iovec*) (intptr_t) (*env)->GetLongField(env, packet, packetMemoryAddressFieldId);
+        msg[i].msg_hdr.msg_iovlen = (*env)->GetIntField(env, packet, packetCountFieldId);
+
+        msg[i].msg_hdr.msg_name = &(addr[i]);
```

Comment:
Do not believe me but I have some kernel guys here that says to use `addr + i` instead because `&(addr[i])`
Not sue that that is undefined too, but it works on any compiler and is more readable so you can let it as it is, your choice bud :) :+1:

Reply:
So basically you suggest something like this ?:

```c
sockaddr_storage* addr_ptr = &addr;
msg[i].msg_hdr.msg_name = addr + i;
```

Reply:
Probably just `msg[i].msg_hdr.msg_name = addr + i;` but yes that's the idea: given that `addr` has type `struct sockaddr_storage` the `+i` will take into account the correct size of struct...
But as I've said above, is just a different way to do that: regardless what my collagues says I don't know of any compiler that break on `&(addr[i])` so feel free to choose :)

Reply:
lets do it